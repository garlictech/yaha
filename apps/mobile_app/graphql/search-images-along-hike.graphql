query SearchAroundHike($query: SearchAroundHikeInput!) {
  searchAroundHike(query: $query) {
    items
    nextToken
    total
  }
}
