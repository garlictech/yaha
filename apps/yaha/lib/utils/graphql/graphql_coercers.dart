import 'package:intl/intl.dart';

final dateFormatter = DateFormat('yyyy.MM.dd');
final timeFormatter = DateFormat('HH:mm:ss');
final timeShortFormatter = DateFormat('HH:mm');
final dateTimeFormatter = DateFormat('yyyy.MM.dd HH:mm');

DateTime fromGraphQLAWSDateTimeToDartDateTime(String date) => DateTime.parse(date).toLocal();
String fromDartDateTimeToGraphQLAWSDateTime(DateTime date) => dateFormatter.format(date);
DateTime? fromGraphQLAWSDateTimeNullableToDartDateTimeNullable(String? date) =>
    date == null ? null : DateTime.parse(date).toLocal();
String? fromDartDateTimeNullableToGraphQLAWSDateTimeNullable(DateTime? date) =>
    date == null ? null : dateFormatter.format(date);
