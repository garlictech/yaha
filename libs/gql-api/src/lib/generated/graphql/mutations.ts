/* tslint:disable */
/* eslint-disable */
// this is an auto generated file. This will be overwritten

export const createHike = /* GraphQL */ `
  mutation CreateHike(
    $input: CreateHikeInput!
    $condition: ModelHikeConditionInput
  ) {
    createHike(input: $input, condition: $condition) {
      id
      description {
        languageKey
        title
        summary
        fullDescription
        type
      }
      location
      publicationState
      featured
      rate
      offlineMap
      featuredStops {
        poiId
        description {
          languageKey
          title
          summary
          fullDescription
          type
        }
      }
      checkpoints {
        id
        location {
          lat
          lon
          ele
        }
        types
        description {
          languageKey
          title
          summary
          fullDescription
          type
        }
        tags
        sourceObject {
          objectType
          languageKey
          objectId
          url
        }
        address
        phoneFloat
        openingHours
      }
      tags
      route {
        distance
        uphill
        downhill
        track
        bigBuffer
        smallBuffer
        averageTime
        score
        difficulty
        bounds {
          NorthEast {
            lat
            lon
            ele
          }
          SouthWest {
            lat
            lon
            ele
          }
        }
        isRoundTrip
        poiSearchBox {
          NorthEast {
            lat
            lon
            ele
          }
          SouthWest {
            lat
            lon
            ele
          }
        }
      }
      startPoint {
        lat
        lon
        ele
      }
      endPoint {
        lat
        lon
        ele
      }
      createdAt
      updatedAt
      _version
      _deleted
      _lastChangedAt
    }
  }
`;
export const updateHike = /* GraphQL */ `
  mutation UpdateHike(
    $input: UpdateHikeInput!
    $condition: ModelHikeConditionInput
  ) {
    updateHike(input: $input, condition: $condition) {
      id
      description {
        languageKey
        title
        summary
        fullDescription
        type
      }
      location
      publicationState
      featured
      rate
      offlineMap
      featuredStops {
        poiId
        description {
          languageKey
          title
          summary
          fullDescription
          type
        }
      }
      checkpoints {
        id
        location {
          lat
          lon
          ele
        }
        types
        description {
          languageKey
          title
          summary
          fullDescription
          type
        }
        tags
        sourceObject {
          objectType
          languageKey
          objectId
          url
        }
        address
        phoneFloat
        openingHours
      }
      tags
      route {
        distance
        uphill
        downhill
        track
        bigBuffer
        smallBuffer
        averageTime
        score
        difficulty
        bounds {
          NorthEast {
            lat
            lon
            ele
          }
          SouthWest {
            lat
            lon
            ele
          }
        }
        isRoundTrip
        poiSearchBox {
          NorthEast {
            lat
            lon
            ele
          }
          SouthWest {
            lat
            lon
            ele
          }
        }
      }
      startPoint {
        lat
        lon
        ele
      }
      endPoint {
        lat
        lon
        ele
      }
      createdAt
      updatedAt
      _version
      _deleted
      _lastChangedAt
    }
  }
`;
export const deleteHike = /* GraphQL */ `
  mutation DeleteHike(
    $input: DeleteHikeInput!
    $condition: ModelHikeConditionInput
  ) {
    deleteHike(input: $input, condition: $condition) {
      id
      description {
        languageKey
        title
        summary
        fullDescription
        type
      }
      location
      publicationState
      featured
      rate
      offlineMap
      featuredStops {
        poiId
        description {
          languageKey
          title
          summary
          fullDescription
          type
        }
      }
      checkpoints {
        id
        location {
          lat
          lon
          ele
        }
        types
        description {
          languageKey
          title
          summary
          fullDescription
          type
        }
        tags
        sourceObject {
          objectType
          languageKey
          objectId
          url
        }
        address
        phoneFloat
        openingHours
      }
      tags
      route {
        distance
        uphill
        downhill
        track
        bigBuffer
        smallBuffer
        averageTime
        score
        difficulty
        bounds {
          NorthEast {
            lat
            lon
            ele
          }
          SouthWest {
            lat
            lon
            ele
          }
        }
        isRoundTrip
        poiSearchBox {
          NorthEast {
            lat
            lon
            ele
          }
          SouthWest {
            lat
            lon
            ele
          }
        }
      }
      startPoint {
        lat
        lon
        ele
      }
      endPoint {
        lat
        lon
        ele
      }
      createdAt
      updatedAt
      _version
      _deleted
      _lastChangedAt
    }
  }
`;
