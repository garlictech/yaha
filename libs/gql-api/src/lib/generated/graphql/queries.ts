/* tslint:disable */
/* eslint-disable */
// this is an auto generated file. This will be overwritten

export const getImage = /* GraphQL */ `
  query GetImage($id: ID!) {
    getImage(id: $id) {
      id
      location {
        lat
        lon
      }
      original {
        url
        width
        height
      }
      card {
        url
        width
        height
      }
      thumbnail {
        url
        width
        height
      }
      sourceObject {
        objectType
        languageKey
        objectId
        url
      }
      attributions
      labels
      tagged
      banned
      createdAt
      updatedAt
    }
  }
`;
export const listImages = /* GraphQL */ `
  query ListImages(
    $filter: ModelImageFilterInput
    $limit: Int
    $nextToken: String
  ) {
    listImages(filter: $filter, limit: $limit, nextToken: $nextToken) {
      items {
        id
        location {
          lat
          lon
        }
        original {
          url
          width
          height
        }
        card {
          url
          width
          height
        }
        thumbnail {
          url
          width
          height
        }
        sourceObject {
          objectType
          languageKey
          objectId
          url
        }
        attributions
        labels
        tagged
        banned
        createdAt
        updatedAt
      }
      nextToken
    }
  }
`;
export const searchImages = /* GraphQL */ `
  query SearchImages(
    $filter: SearchableImageFilterInput
    $sort: [SearchableImageSortInput]
    $limit: Int
    $nextToken: String
    $from: Int
    $aggregates: [SearchableImageAggregationInput]
  ) {
    searchImages(
      filter: $filter
      sort: $sort
      limit: $limit
      nextToken: $nextToken
      from: $from
      aggregates: $aggregates
    ) {
      items {
        id
        location {
          lat
          lon
        }
        original {
          url
          width
          height
        }
        card {
          url
          width
          height
        }
        thumbnail {
          url
          width
          height
        }
        sourceObject {
          objectType
          languageKey
          objectId
          url
        }
        attributions
        labels
        tagged
        banned
        createdAt
        updatedAt
      }
      nextToken
      total
      aggregateItems {
        name
        result {
          ... on SearchableAggregateScalarResult {
            value
          }
          ... on SearchableAggregateBucketResult {
            buckets {
              key
              doc_count
            }
          }
        }
      }
    }
  }
`;
export const getPoi = /* GraphQL */ `
  query GetPoi($id: ID!) {
    getPoi(id: $id) {
      id
      elevation
      location {
        lat
        lon
      }
      type
      description {
        languageKey
        title
        summary
        fullDescription
        type
      }
      tags
      sourceObject {
        objectType
        languageKey
        objectId
        url
      }
      address
      phoneNumber
      openingHours
      createdAt
      updatedAt
    }
  }
`;
export const listPois = /* GraphQL */ `
  query ListPois(
    $filter: ModelPoiFilterInput
    $limit: Int
    $nextToken: String
  ) {
    listPois(filter: $filter, limit: $limit, nextToken: $nextToken) {
      items {
        id
        elevation
        location {
          lat
          lon
        }
        type
        description {
          languageKey
          title
          summary
          fullDescription
          type
        }
        tags
        sourceObject {
          objectType
          languageKey
          objectId
          url
        }
        address
        phoneNumber
        openingHours
        createdAt
        updatedAt
      }
      nextToken
    }
  }
`;
export const searchPois = /* GraphQL */ `
  query SearchPois(
    $filter: SearchablePoiFilterInput
    $sort: [SearchablePoiSortInput]
    $limit: Int
    $nextToken: String
    $from: Int
    $aggregates: [SearchablePoiAggregationInput]
  ) {
    searchPois(
      filter: $filter
      sort: $sort
      limit: $limit
      nextToken: $nextToken
      from: $from
      aggregates: $aggregates
    ) {
      items {
        id
        elevation
        location {
          lat
          lon
        }
        type
        description {
          languageKey
          title
          summary
          fullDescription
          type
        }
        tags
        sourceObject {
          objectType
          languageKey
          objectId
          url
        }
        address
        phoneNumber
        openingHours
        createdAt
        updatedAt
      }
      nextToken
      total
      aggregateItems {
        name
        result {
          ... on SearchableAggregateScalarResult {
            value
          }
          ... on SearchableAggregateBucketResult {
            buckets {
              key
              doc_count
            }
          }
        }
      }
    }
  }
`;
export const getHike = /* GraphQL */ `
  query GetHike($id: ID!) {
    getHike(id: $id) {
      id
      description {
        languageKey
        title
        summary
        fullDescription
        type
      }
      route {
        type
        coordinates
      }
      publicationState
      featured
      rate
      featuredStops {
        poiId
        description {
          languageKey
          title
          summary
          fullDescription
          type
        }
      }
      checkpoints {
        poiId
        point {
          lat
          lon
          elevation
        }
      }
      tags
      closestPlace
      location {
        lat
        lon
      }
      createdAt
      updatedAt
    }
  }
`;
export const listHikes = /* GraphQL */ `
  query ListHikes(
    $filter: ModelHikeFilterInput
    $limit: Int
    $nextToken: String
  ) {
    listHikes(filter: $filter, limit: $limit, nextToken: $nextToken) {
      items {
        id
        description {
          languageKey
          title
          summary
          fullDescription
          type
        }
        route {
          type
          coordinates
        }
        publicationState
        featured
        rate
        featuredStops {
          poiId
          description {
            languageKey
            title
            summary
            fullDescription
            type
          }
        }
        checkpoints {
          poiId
          point {
            lat
            lon
            elevation
          }
        }
        tags
        closestPlace
        location {
          lat
          lon
        }
        createdAt
        updatedAt
      }
      nextToken
    }
  }
`;
export const searchHikes = /* GraphQL */ `
  query SearchHikes(
    $filter: SearchableHikeFilterInput
    $sort: [SearchableHikeSortInput]
    $limit: Int
    $nextToken: String
    $from: Int
    $aggregates: [SearchableHikeAggregationInput]
  ) {
    searchHikes(
      filter: $filter
      sort: $sort
      limit: $limit
      nextToken: $nextToken
      from: $from
      aggregates: $aggregates
    ) {
      items {
        id
        description {
          languageKey
          title
          summary
          fullDescription
          type
        }
        route {
          type
          coordinates
        }
        publicationState
        featured
        rate
        featuredStops {
          poiId
          description {
            languageKey
            title
            summary
            fullDescription
            type
          }
        }
        checkpoints {
          poiId
          point {
            lat
            lon
            elevation
          }
        }
        tags
        closestPlace
        location {
          lat
          lon
        }
        createdAt
        updatedAt
      }
      nextToken
      total
      aggregateItems {
        name
        result {
          ... on SearchableAggregateScalarResult {
            value
          }
          ... on SearchableAggregateBucketResult {
            buckets {
              key
              doc_count
            }
          }
        }
      }
    }
  }
`;
export const searchByContent = /* GraphQL */ `
  query SearchByContent($query: SearchByContentInput!) {
    searchByContent(query: $query) {
      items
      nextToken
      total
    }
  }
`;
export const searchByRadius = /* GraphQL */ `
  query SearchByRadius($query: SearchByRadiusInput!) {
    searchByRadius(query: $query) {
      items
      nextToken
      total
    }
  }
`;
export const searchHikeByRadius = /* GraphQL */ `
  query SearchHikeByRadius($query: SearchHikeByRadiusInput!) {
    searchHikeByRadius(query: $query) {
      items
      nextToken
      total
    }
  }
`;
export const searchInEnvelope = /* GraphQL */ `
  query SearchInEnvelope($query: SearchInEnvelopeInput!) {
    searchInEnvelope(query: $query) {
      items
      nextToken
      total
    }
  }
`;
export const searchInShape = /* GraphQL */ `
  query SearchInShape($query: SearchInShapeInput!) {
    searchInShape(query: $query) {
      items
      nextToken
      total
    }
  }
`;
export const searchInMultipolygon = /* GraphQL */ `
  query SearchInMultipolygon($query: SearchInMultipolygonInput!) {
    searchInMultipolygon(query: $query) {
      items
      nextToken
      total
    }
  }
`;
export const searchAroundHike = /* GraphQL */ `
  query SearchAroundHike($query: SearchAroundHikeInput!) {
    searchAroundHike(query: $query) {
      items
      nextToken
      total
    }
  }
`;
export const searchSafeImagesAroundHike = /* GraphQL */ `
  query SearchSafeImagesAroundHike($query: SearchSafeImagesAroundHikeInput!) {
    searchSafeImagesAroundHike(query: $query) {
      items
      nextToken
      total
    }
  }
`;
export const searchSafeImagesAroundLocation = /* GraphQL */ `
  query SearchSafeImagesAroundLocation($query: SearchByRadiusInput!) {
    searchSafeImagesAroundLocation(query: $query) {
      items
      nextToken
      total
    }
  }
`;
