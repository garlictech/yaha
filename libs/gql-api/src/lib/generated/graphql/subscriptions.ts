/* tslint:disable */
/* eslint-disable */
// this is an auto generated file. This will be overwritten

export const onCreateImage = /* GraphQL */ `
  subscription OnCreateImage {
    onCreateImage {
      id
      location {
        lat
        lon
      }
      original {
        url
        width
        height
      }
      card {
        url
        width
        height
      }
      thumbnail {
        url
        width
        height
      }
      sourceObject {
        objectType
        languageKey
        objectId
        url
      }
      attributions
      labels
      tagged
      banned
      createdAt
      updatedAt
    }
  }
`;
export const onUpdateImage = /* GraphQL */ `
  subscription OnUpdateImage {
    onUpdateImage {
      id
      location {
        lat
        lon
      }
      original {
        url
        width
        height
      }
      card {
        url
        width
        height
      }
      thumbnail {
        url
        width
        height
      }
      sourceObject {
        objectType
        languageKey
        objectId
        url
      }
      attributions
      labels
      tagged
      banned
      createdAt
      updatedAt
    }
  }
`;
export const onDeleteImage = /* GraphQL */ `
  subscription OnDeleteImage {
    onDeleteImage {
      id
      location {
        lat
        lon
      }
      original {
        url
        width
        height
      }
      card {
        url
        width
        height
      }
      thumbnail {
        url
        width
        height
      }
      sourceObject {
        objectType
        languageKey
        objectId
        url
      }
      attributions
      labels
      tagged
      banned
      createdAt
      updatedAt
    }
  }
`;
export const onCreatePoi = /* GraphQL */ `
  subscription OnCreatePoi {
    onCreatePoi {
      id
      elevation
      location {
        lat
        lon
      }
      type
      description {
        languageKey
        title
        summary
        fullDescription
        type
      }
      tags
      sourceObject {
        objectType
        languageKey
        objectId
        url
      }
      address
      phoneNumber
      openingHours
      createdAt
      updatedAt
    }
  }
`;
export const onUpdatePoi = /* GraphQL */ `
  subscription OnUpdatePoi {
    onUpdatePoi {
      id
      elevation
      location {
        lat
        lon
      }
      type
      description {
        languageKey
        title
        summary
        fullDescription
        type
      }
      tags
      sourceObject {
        objectType
        languageKey
        objectId
        url
      }
      address
      phoneNumber
      openingHours
      createdAt
      updatedAt
    }
  }
`;
export const onDeletePoi = /* GraphQL */ `
  subscription OnDeletePoi {
    onDeletePoi {
      id
      elevation
      location {
        lat
        lon
      }
      type
      description {
        languageKey
        title
        summary
        fullDescription
        type
      }
      tags
      sourceObject {
        objectType
        languageKey
        objectId
        url
      }
      address
      phoneNumber
      openingHours
      createdAt
      updatedAt
    }
  }
`;
export const onCreateHike = /* GraphQL */ `
  subscription OnCreateHike {
    onCreateHike {
      id
      description {
        languageKey
        title
        summary
        fullDescription
        type
      }
      route {
        type
        coordinates
      }
      publicationState
      featured
      rate
      featuredStops {
        poiId
        description {
          languageKey
          title
          summary
          fullDescription
          type
        }
      }
      checkpoints {
        poiId
        point {
          lat
          lon
          elevation
        }
      }
      tags
      closestPlace
      location {
        lat
        lon
      }
      createdAt
      updatedAt
    }
  }
`;
export const onUpdateHike = /* GraphQL */ `
  subscription OnUpdateHike {
    onUpdateHike {
      id
      description {
        languageKey
        title
        summary
        fullDescription
        type
      }
      route {
        type
        coordinates
      }
      publicationState
      featured
      rate
      featuredStops {
        poiId
        description {
          languageKey
          title
          summary
          fullDescription
          type
        }
      }
      checkpoints {
        poiId
        point {
          lat
          lon
          elevation
        }
      }
      tags
      closestPlace
      location {
        lat
        lon
      }
      createdAt
      updatedAt
    }
  }
`;
export const onDeleteHike = /* GraphQL */ `
  subscription OnDeleteHike {
    onDeleteHike {
      id
      description {
        languageKey
        title
        summary
        fullDescription
        type
      }
      route {
        type
        coordinates
      }
      publicationState
      featured
      rate
      featuredStops {
        poiId
        description {
          languageKey
          title
          summary
          fullDescription
          type
        }
      }
      checkpoints {
        poiId
        point {
          lat
          lon
          elevation
        }
      }
      tags
      closestPlace
      location {
        lat
        lon
      }
      createdAt
      updatedAt
    }
  }
`;
