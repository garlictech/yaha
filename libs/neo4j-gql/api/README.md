# neo4j-gql-api

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test neo4j-gql-api` to execute the unit tests via [Jest](https://jestjs.io).
