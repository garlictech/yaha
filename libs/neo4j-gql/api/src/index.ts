export * from './lib/neo4j-gql-api';
