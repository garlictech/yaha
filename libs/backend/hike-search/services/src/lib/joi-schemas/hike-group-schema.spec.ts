import { PublicationState } from '../api/graphql';
import {
  isCreateHikeGroupInput,
  isHikeGroup,
  validateCreateHikeGroupInput,
  validateHikeGroup,
} from './hike-group-schema';

describe('Test the hike group schema', () => {
  it('it should reconstruct the HikeGroup', async () => {
    const hikeGroup = {
      id: 'bd1e9454-aeb5-488f-9e0b-1a5b512fe1bf',
      description: [
        {
          languageKey: 'en_US',
          title: 'TITLE',
          type: 'html',
        },
      ],
      publicationState: PublicationState.draft,
    };

    expect(await validateHikeGroup(hikeGroup).toPromise()).toMatchSnapshot();
    expect(isHikeGroup(hikeGroup)).toEqual(true);
  });

  it('it should report the wrong fields', done => {
    const hikeGroup = {
      id: '',
      description: {
        languageKey: 'en_US',
      },
    };

    validateHikeGroup(hikeGroup).subscribe(undefined, err => {
      expect(err).toMatchSnapshot();
      expect(isHikeGroup(hikeGroup)).toEqual(false);
      done();
    });
  });
});

describe('Test the create hike group input schema', () => {
  it('it should reconstruct the CreateHikeGroupInput', async () => {
    const hikeGroup = {
      description: [
        {
          languageKey: 'en_US',
          title: 'The title',
          type: 'markdown',
        },
      ],
      hikeIds: ['hikeid-1'],
      publicationState: PublicationState.draft,
    };

    expect(
      await validateCreateHikeGroupInput(hikeGroup).toPromise(),
    ).toMatchSnapshot();
    expect(isCreateHikeGroupInput(hikeGroup)).toEqual(true);
  });

  it('it should report the wrong fields', done => {
    const hikeGroup = {
      id: '',
      description: {
        languageKey: 'en_US',
      },
    };

    validateCreateHikeGroupInput(hikeGroup).subscribe(undefined, err => {
      expect(err).toMatchSnapshot();
      expect(isCreateHikeGroupInput(hikeGroup)).toEqual(false);
      done();
    });
  });
});
