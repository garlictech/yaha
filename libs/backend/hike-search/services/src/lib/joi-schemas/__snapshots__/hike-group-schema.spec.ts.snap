// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the create hike group input schema it should reconstruct the CreateHikeGroupInput 1`] = `
Object {
  "description": Array [
    Object {
      "languageKey": "en_US",
      "title": "The title",
      "type": "markdown",
    },
  ],
  "hikeIds": Array [
    "hikeid-1",
  ],
  "publicationState": "draft",
}
`;

exports[`Test the create hike group input schema it should report the wrong fields 1`] = `
Array [
  Object {
    "exception": "any.required",
    "message": "\\"publicationState\\" is required",
    "path": Array [
      "publicationState",
    ],
  },
  Object {
    "exception": "array.base",
    "message": "\\"description\\" must be an array",
    "path": Array [
      "description",
    ],
  },
  Object {
    "exception": "object.allowUnknown",
    "message": "\\"id\\" is not allowed",
    "path": Array [
      "id",
    ],
  },
]
`;

exports[`Test the hike group schema it should reconstruct the HikeGroup 1`] = `
Object {
  "description": Array [
    Object {
      "languageKey": "en_US",
      "title": "TITLE",
      "type": "html",
    },
  ],
  "id": "bd1e9454-aeb5-488f-9e0b-1a5b512fe1bf",
  "publicationState": "draft",
}
`;

exports[`Test the hike group schema it should report the wrong fields 1`] = `
Array [
  Object {
    "exception": "any.empty",
    "message": "\\"id\\" is not allowed to be empty",
    "path": Array [
      "id",
    ],
  },
  Object {
    "exception": "any.required",
    "message": "\\"publicationState\\" is required",
    "path": Array [
      "publicationState",
    ],
  },
  Object {
    "exception": "array.base",
    "message": "\\"description\\" must be an array",
    "path": Array [
      "description",
    ],
  },
]
`;
